docs/overview.md

```diff
@@ -1,59 +1,151 @@
+{"digest":"sha256:42d56485c1f672e394a02855048774621731c8fd44a54dc816a421a3a52b8482","size":10047608,"fetchedSize":2047608,"fetchedPercent":20.379059374131632}
+{"digest":"sha256:6a5826d877de5c93fb4a9e1d0369cfdef6d43df2610562501ebf42e4bcb2ef73","size":54352828,"fetchedSize":2302828,"fetchedPercent":4.236813584014432}
+{"digest":"sha256:a4d35801573274df19d9c2ae2aed80eba96d5aa69a38c464e1f01f9abf81e34e","size":70359295,"fetchedSize":2259295,"fetchedPercent":3.211082487395588}
+{"digest":"sha256:ab13100112faac6e04d2da2281db3df942efc8cef2532ab2cac688c6232944d8","size":7890588,"fetchedSize":2140588,"fetchedPercent":27.12837116828302}
+{"digest":"sha256:e8cc31024eb09fe216ad906392aec139038330c6d29dfd3fe5c81c4b2dd21430","size":52934435,"fetchedSize":2634435,"fetchedPercent":4.976788738748227}
+{"digest":"sha256:f077511be7d385c17ba88980379c5cd0aab7068844dffa7a1cefbf68cc3daea3","size":580,"fetchedSize":580,"fetchedPercent":100}
+```
+
+## Registry-related configuration
+
+You can configure stargz snapshotter for accessing registries with custom configurations.
+The config file must be formatted with TOML and can be passed to stargz snapshotter with `--config` option.
+
+### Authentication
+
+Stargz snapshotter doesn't share private registries creds with containerd.
+Instead, this supports authentication in the following methods,
 
+- Using `DOCKER_CONFIG` or `~/.docker/config.json`
+- Using Kubernetes secrets (type = `kubernetes.io/dockerconfigjson`)
+
+By default, This snapshotter tries to get creds from `DOCKER_CONFIG` or `~/.docker/config.json`.
+Following example enables stargz snapshotter to access to private registries using `docker login` command.
+Stargz snapshotter doesn't share credentials with containerd so credentials specified by `ctr-remote`'s `--user` option in the example is just for containerd.
+
+```console
+# docker login
+(Enter username and password)
+# ctr-remote image rpull --user <username>:<password> docker.io/<your-repository>/ubuntu:18.04
 ```
-# ctr-remote image optimize --plain-http --entrypoint='[ "/bin/bash", "-c" ]' --args='[ "ls" ]' \
-             ubuntu:18.04 http://registry2:5000/ubuntu:18.04
+
+Following configuration enables stargz snapshotter to access to private registries using kubernetes secrets (type = `kubernetes.io/dockerconfigjson`) in the cluster using kubeconfig files.
+You can specify the path of kubeconfig file using `kubeconfig_path` option.
+It's no problem that the specified file doesn't exist when this snapshotter starts.
+In this case, snapsohtter polls the file until actually provided.
+This is useful for some environments (e.g. single node cluster with containerized apiserver) where stargz snapshotter needs to start before everything, including booting containerd/kubelet/apiserver and configuring users/roles.
+If no `kubeconfig_path` is specified, snapshotter searches kubeconfig files from `KUBECONFIG` or `~/.kube/config`.
+
+```toml

```

Comment:
The file path?

Reply:
Could you elaborate this comment? Should we describe the file path that this config is written to?

Reply:
yes for clarity

Reply:
Thanks for the feedback.
`containerd-stargz-grpc` currently doesn't have the default config file path. 
So I added the description `The config file can be passed to stargz snapshotter using containerd-stargz-grpc's --config option.`.